File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void) {
    // Get money from user
    float money;
    do {
        money = get_float("Change owed: ");
    } while (money < 0);
    int cents = round(money * 100);
    
    // Counting algorithm
    int numberOfCoins = 0;
    while (cents != 0) {
        if (cents >= 25) {
            cents -= 25;
        } else if (cents >= 10) {
            cents -= 10;
        } else if (cents >= 5) {
            cents -= 5;
        } else {
            cents -= 1;
        };
        numberOfCoins++;
    }
    printf("%i\n", numberOfCoins);
}

File: pset6/cash/cash.py
from cs50 import get_float

def main():
    # Prompt the user for monetary amount
    change = 0
    while True:
        change = get_float("Change owed: ")
        if (change >= 0):
            break
    # Convert to cents because of floating point errors
    changeCents = int(change * 100)
    numberOfCoins = 0
    # Loop through and keep deducting cents until it is okay
    while changeCents > 0:
        changeCents -= deductCents(changeCents)
        numberOfCoins += 1
    print(numberOfCoins)

def deductCents(money):
    if money >= 25:
        return 25
    elif money >= 10:
        return 10
    elif money >= 5:
        return 5
    else:
        return 1
main()